File: benchmarks/listsum.py
def build(n):
    a = None
    while n: a = (n, a); n -= 1
    return a
def suml(xs):
    acc = 0
    while xs is not None: acc += xs[0]; xs = xs[1]
    return acc
print(suml(build(20000000)))

File: benchmarks/listsum.js
function build(n){ let a = null; while (n){ a = {v:n, t:a}; n--; } return a; }
function suml(xs){ let acc = 0; while (xs){ acc += xs.v; xs = xs.t; } return acc; }
console.log(suml(build(20000000)));
